contracts/QuestFactory.sol

```diff
@@ -76,7 +76,7 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
         uint256 rewardAmountOrTokenId_,
         string memory contractType_,
         string memory questId_
-    ) public onlyRole(CREATE_QUEST_ROLE) returns (address) {
+    ) external payable onlyRole(CREATE_QUEST_ROLE) returns (address) {
```

Comment:
Why introduce `payable`? Don't think anyone ever needs to send ETH to create a quest

Reply:
If a function modifier such as onlyOwner is used, the function will revert if a normal user tries to pay the function. Marking the function as payable will lower the gas cost for legitimate callers because the compiler will not include checks for whether a payment was provided. 

The extra opcodes avoided are
CALLVALUE(2),DUP1(3),ISZERO(3),PUSH2(3),JUMPI(10),PUSH1(3),DUP1(3),REVERT(0),JUMPDEST(1),POP(2), which costs an average of about 21 gas per call to the function, in addition to the extra deployment cost.